docker-app/qfieldcloud/core/models.py

```diff
@@ -1083,6 +1083,17 @@ class PackagingOffliner(models.TextChoices):
         QGISCORE = "qgiscore", _("QGIS Core Offline Editing (deprecated)")
         PYTHONMINI = "pythonmini", _("Optimized Packager")
 
+    def get_localized_layers(self) -> list[dict]:
+        """
+        Returns a list of all localized layers (as dictionaries)
+        from project_details['layers_by_id'].
+        """
+        layers = (
+            self.project_details.get("layers_by_id", {}) if self.project_details else {}
+        )
+
+        return list(filter(lambda ld: ld.get("is_localized", False), layers.values()))
```

Comment:
```suggestion
        return filter(lambda ld: ld.get("is_localized", False), layers.values())
```
When you said "it works better with serializers" and "avoid any surprises", what did you mean?

Reply:
`filter` lead to issues such as exhausted iterators. For example, if you return a filter object and then iterate over it twice, (e.g. once in a serializer's validation step and again in its representation step), then the first iteration will consume the iterator leading to an empty list during the second iteration.

Reply:
This is true, but what you have described is a property of any iterator, isn't it? I would leave the decision what to do with a returned iterator to the consumer of a method, rather than adding a performance penalty for consuming an iterator by default.

But if you find this confusing, I can live with the list, but in that case would be better if we rewrite it to:

```
localized_layers_data: list[LayerDataDict] = []

for layer_data in layers_by_id.values():
    if not layer_data.get("is_localized", False):
        continue

    localized_layers_data.append(layer_data)

return localized_layers_data
```

More imperative, more better.

Reply:
@tdrivas can we close this comment?

Reply:
Sure, last suggestion has been applied!